layouts.h

```diff
@@ -12,5 +12,11 @@ void overviewlayout(Monitor *m);
 void tcl(Monitor * m);
 void tile(Monitor *m);
 void floatl(Monitor *m);
-
+static inline Client* findVisibleClient(Client *c){
+	for (Client* client = c; client ; client = client->next){
```

Comment:
Please move the declaration of `c` back out of the for loop. I know "newer" C standards allow it, but it's done nowhere else in the code. Let's keep the style consistent and don't break compiler compatibility over this.

Reply:
I maintaining a consistent style. But what compatibility issues are there? DWM and instantWM both use C99 which is when that loop feature was introduced. Is there a reason to support pre-99 compiling?

Reply:
It's just style consistency here. C devs don't like it too much, when their C looks like Java, C# or C++ :wink: 
Just unexpected and stops you in your tracks when reading, because it's the odd one out.

Reply:
I fixed it. However, I am still going to argue for moving to the C99 (and later) style. I will be more than happy to implement the change as part of a bigger PR. That would take the style argument off the table. 

**Option A:**
```
int bar; 
for(bar = 0; bar < baz; ++bar){....}
```

_NB_: results in bar being undefined until the loop, the code has a higher chance of introducing a bug as a result.  

**Option B:**
```
int bar = .... ; // some default value 
for(bar = 0; bar < baz; ++bar){....}
```

_NB_: Fixes the problem with A, but results in a needless assignment being done. The compiler _may_ fix this. 

**Option C:**
```
int bar = 0; 
for(; bar < baz; ++bar){....}
```

_NB_: Why not just use a while loop at this point. 

**Option D:**
```
for(int bar = 0; bar < baz; ++bar){....}
```

_NB_: Benefits include:
1. Minimizes the scope of the variable (This is good for lots of reasons)
2. Represents a modern style, making the code more accessible to a wider audience. 
3. Decreases SLOC
4. Helps to ensure the creation and initialization happens together. 
5. Makes it clear that the var is only used in that scope
6. Makes the loop more portable. 

Once you remove that backwards compatibility and consistency argument, I don't think there is another good one. Also, I will admit, to some degree this is trivial, but there are reasons C99 added the feature 20+ years ago as well as most coding standards recommending it.

Reply:
If you do it, do it everywhere. End of debate.